docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,663 @@
+  }
+go();
+```
+
+We need to remember to set `Lit.Actions.setResponse` before we finish executing the Lit Action. This defines the response that will be shown when the Lit Action finishes executing.
+
+The complete swap Lit Action can be found [here](https://github.com/anshss/lit-evm-swap/blob/main/lit/swapAction.js).
+
+## 3) Implementing Mint/Grant/Burn with SDK
+
+### Installing and importing the relevant packages
+
+```bash
+npm i @lit-protocol/lit-node-client @lit-protocol/contracts-sdk 
+@simplewebauthn/browser  @lit-protocol/types ethers
+```
+
+
+```js
+import { LitNodeClient } from "@lit-protocol/lit-node-client";
+import { LitContracts } from "@lit-protocol/contracts-sdk";
+import { LitNetwork, AuthMethodScope, LIT_CHAINS } from "@lit-protocol/constants";
+import { LitAbility } from "@lit-protocol/types";
+import {
+    LitActionResource,
+    createSiweMessageWithRecaps,
+    generateAuthSig,
+    LitPKPResource,
+} from "@lit-protocol/auth-helpers";
+import { ethers } from "ethers";
+import bs58 from "bs58";
+```
+
+### Creating our two wallet instances
+
+These wallets will represent the two parties involved in the swap.
+
+```js
+async function getWalletA() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_1,
+        provider
+    );
+    return wallet;
+}
+
+async function getWalletB() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_2,
+        provider
+    );
+    return wallet;
+}
+```
+
+### Creating our Lit Action and uploading it to IPFS
+
+Once we are done with our Lit Action, we need to upload it to IPFS so we get an immutable CID which will always point to our action.
+
+```js
+export async function createLitAction() {
+    console.log("creating lit action..");
+    const action = createERC20SwapLitAction(chainAParams, chainBParams);
+    const ipfsCid = await uploadViaPinata(action);
+
+    console.log("Lit Action code:\n", action);
+    console.log("IPFS CID: ", ipfsCid);
+    return ipfsCid;
+}
+```
+
+### Create a Mint/Grant/Burn PKP
+
+:::info
+Currently, direct calls to the smart contract’s m/g/b function are currently failing with the Lit Network. We’ll replicate the same functionality using the SDK.
+:::
+
+We'll create a single function to perform 3 calls at once
+
+- Mint a PKP with the user's wallet
+- Add the Lit Action as a permitted auth Method
+- Burn the PKP NFT by transferring it to itself so auth methods become immutable
+
+Now, our PKP will only be used to execute the Lit Action we permitted. It will generate signatures only when the Lit Action is running and not in any other situations. We have already designed the Lit Action to create the correct signed transactions based on the conditions.
```

Comment:
> Now, our PKP will only be used to execute the Lit Action we permitted. It will generate signatures only when the Lit Action is running and not in any other situations. We have already designed the Lit Action to create the correct signed transactions based on the conditions.

"Now our PKP can only be used to execute the Lit Action we permitted when we added it as an auth method. It will only generate signatures when the Lit Action is running and the conditions we set have been met"